Userland/Libraries/LibCrypto/Hash/SHA1.cpp

```diff
@@ -77,13 +77,16 @@ static void transform_impl_base(u32 (&state)[5], u8 const (&data)[64])
     secure_zero(blocks, 16 * sizeof(u32));
 }
 
-#if (ARCH(I386) || ARCH(X86_64))
 // Note: The SHA extension was introduced with
 //       Intel Goldmont (SSE4.2), Ice Lake (AVX512), Rocket Lake (AVX512), and AMD Zen (AVX2)
 //       So it's safe to assume that if we have SHA we have at least SSE4.2
 //      ~https://en.wikipedia.org/wiki/Intel_SHA_extensions
-[[gnu::target("sha,sse4.2")]] static void transform_impl_sha1(u32 (&state)[5], u8 const (&data)[64])
+#if AK_CAN_CODEGEN_FOR_X86_SHA && AK_CAN_CODEGEN_FOR_X86_SSE42
+template<>
+[[gnu::target("sha,sse4.2")]] void transform_impl<CPUFeature::X86_SSE42 | CPUFeature::X86_SHA>(u32 (&state)[5], u8 const (&data)[64])
 {
+#    define SAME_TARGET gnu::target("sha"), gnu::always_inline
```

Comment:
this seems to drop the `sse4.2` target
although it is not directly used by the target functions it's nice to be consistent
Not sure why that attribute is not inherited anyway

Reply:
Also, `SAME_TARGET` seems pretty undescriptive. Wouldn't `SHA_TARGET` be better? Or am I misunderstanding what this is supposed to mean?

Reply:
Renamed to `SHA_TARGET` and added a bit of discussion (in the commit desc) about why sse4.2 will still be used in these functions even without explicitly stating it as a target.